include/irr/video/IGPUSpecializedShader.h

```diff
@@ -2,14 +2,22 @@
 #define __IRR_I_GPU_SPECIALIZED_SHADER_H_INCLUDED__
 
 #include "irr/core/IReferenceCounted.h"
+#include "irr/asset/ShaderCommons.h"
 
 namespace irr { namespace video
 {
 
 class IGPUSpecializedShader : public core::IReferenceCounted
 {
+public:
+    IGPUSpecializedShader(asset::E_SHADER_STAGE _stage) : m_stage(_stage) {}
+
+    asset::E_SHADER_STAGE getStage() const { return m_stage; }
+
 protected:
     virtual ~IGPUSpecializedShader() = default;
+
+    asset::E_SHADER_STAGE m_stage;
```

Comment:
isnt this in specialization info already?

Reply:
it is but GPU specialized shader doesnt hold its specialization data